src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -206,8 +206,11 @@ public bool TryGetValues(string name, out IEnumerable<string> values)
             return TryGetValues(descriptor, out values);
         }
 
+        protected virtual void EnsureHeaders() { }
+
         internal bool TryGetValues(HeaderDescriptor descriptor, out IEnumerable<string> values)
         {
+            EnsureHeaders();
```

Comment:
This is now adding a virtual method call on a bunch of code paths.  What does that do to performance?

Reply:
Hm. I do not know this. And I do not know what and how should I measure.